addons/sale_project/models/project_project.py

```diff
@@ -175,7 +176,7 @@ def action_view_sos(self):
                 "show_sale": True,
                 'default_partner_id': self.partner_id.id,
                 'default_analytic_account_id': self.analytic_account_id.id,
-                "create_for_project_id": self.id if embedded_action_context else False,
+                "create_for_project_id": create_for_project_id if not embedded_action_context else False,
```

Comment:
2 questions:
- can't you use that variable for the "create" key of the context?
- shouldn't we default on `self.id`? not sure I understand

Reply:
> can't you use that variable for the "create" key of the context?

Nope `create_for_project_id` key is to make sure the SO created contains at least one service (and also set the first service inside that SO to set it on the project stored in that context key).

See https://github.com/odoo/odoo/blob/a16e2c0245ca1199bb4c45d28575aa36ae6bc66b/community/addons/sale_project/models/sale_order.py#L305